tools/PI/DevHome.PI/ViewModels/AppDetailsPageViewModel.cs

```diff
@@ -121,4 +136,123 @@ private void RunAsAdmin()
             CommonHelper.RunAsAdmin(_targetProcess.Id, nameof(AppDetailsPageViewModel));
         }
     }
+
+    private void GetPackageInfo(ProcessDiagnosticInfo pdi)
+    {
+        if (pdi.IsPackaged)
+        {
+            AppInfo.IsPackaged = true;
+            ProcessPackageVisibility = Visibility.Visible;
+
+            var package = pdi.GetAppDiagnosticInfos().FirstOrDefault()?.AppInfo.Package;
+            if (package is not null)
+            {
+                if (package.Id is not null)
+                {
+                    AppInfo.PackageInfo.FullName = package.Id.FullName;
+                    var version = package.Id.Version;
+                    AppInfo.PackageInfo.Version = $"{version.Major}.{version.Minor}.{version.Build}.{version.Revision}";
+                }
+
+                AppInfo.PackageInfo.DisplayName = package.DisplayName;
+                AppInfo.PackageInfo.InstalledDate = package.InstalledDate.ToString(CultureInfo.CurrentCulture);
+                AppInfo.PackageInfo.InstalledPath = package.InstalledPath;
+                AppInfo.PackageInfo.Publisher = package.PublisherDisplayName;
+                AppInfo.PackageInfo.IsDevelopmentMode = package.IsDevelopmentMode;
+                AppInfo.PackageInfo.SignatureKind = $"{package.SignatureKind}";
+                AppInfo.PackageInfo.Status = GetPackageStatus(package);
+
+                List<string> dependencies = [];
+                foreach (var d in package.Dependencies)
+                {
+                    dependencies.Add(d.Id.FullName);
+                }
+
+                AppInfo.PackageInfo.Dependencies = string.Join(", ", dependencies);
+            }
+        }
+        else
+        {
+            ProcessPackageVisibility = Visibility.Collapsed;
+        }
+    }
+
+    private string GetPackageStatus(Package p)
+    {
+        // Convert the individual bool Status properties to a list of matching strings.
+        List<string> trueProperties = [];
+        var status = p.Status;
+        string combinedStatus;
+
+        if (status.DataOffline)
```

Comment:
I am extremely annoyed that package.Status can not be enumerated.  It can't extend to .ToList or .ToDictionary to make this method simpler.